pkg/agent/storage/https.go

```diff
@@ -123,7 +128,10 @@ func (h *HTTPSDownloader) extractHeaders() (map[string]string, error) {
 	var headers map[string]string
 	hostname := h.Uri.Hostname()
 	headerJSON := os.Getenv(hostname + HEADER_SUFFIX)
-	json.Unmarshal([]byte(headerJSON), &headers)
+	err := json.Unmarshal([]byte(headerJSON), &headers)
+	if err != nil {
+		log.Error(err, "failed to unmarshal headers")
```

Comment:
need to return err here

Reply:
This was intentionally not returned since there are existing test cases that assume we still download the model even when there are errors during unmarshaling. I didn't want to break the existing logic so I only logged the error instead returning it.